hy/cmdline.py

```diff
@@ -366,26 +367,30 @@ def hy2py_main():
     import platform
     module_name = "<STDIN>"
 
-    options = dict(prog="hy2py", usage="%(prog)s [options] FILE",
+    options = dict(prog="hy2py", usage="%(prog)s [options] [FILE]",
                    formatter_class=argparse.RawDescriptionHelpFormatter)
     parser = argparse.ArgumentParser(**options)
+    parser.add_argument("FILE", type=str, nargs='?',
+                        help="Input Hy code (use STDIN if \"-\" or "
+                             "not provided)")
     parser.add_argument("--with-source", "-s", action="store_true",
                         help="Show the parsed source structure")
     parser.add_argument("--with-ast", "-a", action="store_true",
                         help="Show the generated AST")
     parser.add_argument("--without-python", "-np", action="store_true",
                         help=("Do not show the Python code generated "
                               "from the AST"))
-    parser.add_argument('args', nargs=argparse.REMAINDER,
-                        help=argparse.SUPPRESS)
 
     options = parser.parse_args(sys.argv[1:])
 
-    if not options.args:
-        parser.exit(1, parser.format_help())
+    stdin_text = None
+    if not options.FILE or options.FILE == '-':
```

Comment:
`if options.FILE is None`

Otherwise, this will end up reading from stdin with the empty string as an argument.